frontend/src/components/programsVacancies/ManageProgramsVacancies.vue

```diff
@@ -0,0 +1,349 @@
+              <v-row>
+                <v-col cols="auto"><v-radio label="Yes" :value="true" /></v-col>
+                <v-col cols="auto"><v-radio label="No" :value="false" /></v-col>
+              </v-row>
+            </v-radio-group>
+          </div>
+        </v-card>
+      </v-row>
+    </v-skeleton-loader>
+  </v-container>
+</template>
+
+<script>
+import AppButton from '@/components/guiComponents/AppButton.vue';
+
+import alertMixin from '@/mixins/alertMixin.js';
+import globalMixin from '@/mixins/globalMixin.js';
+
+import ProgramsVacanciesService from '@/services/programsVacancies.js';
+
+import {
+  AGE_GROUPS,
+  DAYS_OPTIONS,
+  EMPTY_PLACEHOLDER,
+  INDIGENOUS_LED_OPTIONS,
+  LANGUAGE_OPTIONS,
+  MEAL_OPTIONS,
+  PICKUP_OPTIONS,
+  PRESCHOOL_OPTIONS,
+  VACANCY_FIELDS,
+} from '@/utils/constants.js';
+import { formatUTCDate } from '@/utils/format';
+import rules from '@/utils/rules.js';
+
+export default {
+  name: 'ManageProgramsVacancies',
+  components: { AppButton },
+  mixins: [alertMixin, globalMixin],
+  data() {
+    return {
+      isLoading: false,
+      isEditing: false,
+      isProcessing: false,
+      programVacancies: null,
+      form: {},
+      ageGroups: AGE_GROUPS,
+      daysOptions: DAYS_OPTIONS,
+      languageOptions: LANGUAGE_OPTIONS,
+      indigenousLedOptions: INDIGENOUS_LED_OPTIONS,
+      preschoolOptions: PRESCHOOL_OPTIONS,
+      mealOptions: MEAL_OPTIONS,
+      pickupOptions: PICKUP_OPTIONS,
+      vacancyFields: VACANCY_FIELDS,
+    };
+  },
+  computed: {
+    lastUpdated() {
+      return formatUTCDate(this.programVacancies?.updatedOn);
+    },
+  },
+  created() {
+    this.EMPTY_PLACEHOLDER = EMPTY_PLACEHOLDER;
+    this.rules = rules;
+    this.loadData();
+  },
+  methods: {
+    formatUTCDate,
+    parseNumberList(str) {
+      if (!str) return [];
+      return str.split(',').map((x) => Number(x.trim()));
+    },
+    async loadData() {
+      this.isLoading = true;
+      try {
+        const response = await ProgramsVacanciesService.getProgramsVacancies(this.$route.params.facilityId);
+        this.programVacancies = response[0];
+        this.form = {
+          ...this.programVacancies,
+          selectedPrograms: this.parseNumberList(this.programVacancies.selectedPrograms),
+          daysOfOperation: this.parseNumberList(this.programVacancies.daysOfOperation),
+          additionalLanguages: this.parseNumberList(this.programVacancies.additionalLanguages),
+          indigenousLed: this.parseNumberList(this.programVacancies.indigenousLed),
+          preschoolServices: this.parseNumberList(this.programVacancies.preschoolServices),
+        };
+      } catch (error) {
+        this.setFailureAlert('Failed to load Programs and Vacancies.', error);
+      } finally {
+        this.isLoading = false;
+      }
+    },
+    onEdit() {
+      this.isEditing = true;
+    },
+    onCancel() {
+      this.isEditing = false;
+      this.loadData();
```

Comment:
Instead of reloading the data on cancel, we should store a copy of the loaded data and update the current state from that.